### PR TITLE
feat: OAS validation, better error handling, action debugging

### LINKED ISSUE #23: Sync action silently fails with a TypeError & UnhandledPromiseRejectionWarning
The action appears to succeed with a green check, but if you examine the output it is failing with a TypeError.
```
Run readmeio/github-readme-sync@2.0.0
(node:2493) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'match' of undefined
    at /home/runner/work/_actions/readmeio/github-readme-sync/2.0.0/main.js:95:28
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:2493) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2493) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This was is the relevant section of the action config:
```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
      - uses: readmeio/github-readme-sync@2.0.0
        with:
          readme-oas-key: xxxx
           
          # OPTIONAL CONFIG, use if necessary
          oas-file-path: './dev-ops-api/api/swagger.yaml'
          # api-version: 'v1.0.0'
```

### PR DESCRIPTION
**Pull request by kanadgupta**
#### ... A lot! So here are the highlights:
- [x] feat: validate OAS before bundling (5620362)
  - This validates the OAS before we send it via API. Not sure if this will be controversial, maybe we can add this behind a flag so it isn't breaking? Open for discussion on this.
- [x] feat: add debugging (fe5555d)
  - Utilizes GitHub's [Action Debugger](https://github.com/actions/toolkit/blob/master/docs/action-debugging.md) so customers have a little more info to give us when they see issues
- [x] fix: properly parse our error messages (0a230bd)
  - Actually handles our new error message structure, closes #23
- [x] fix: error out if no file is found (c1358ac)
  - Adds an error case for if no file is found

#### As well as some housekeeping:
- [x] ci: add prettier to package.json (4297748)
- [x] ci: add back warnings for console.logs() (1705bf7)
- [x] docs: update some log messages (17c145a)
- [x] refactor: remove an unnecessary if block (7ba10ad)

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -39,21 +48,27 @@ async function run() {
     base: baseFile,
     ignoreErrors: true,
   })
-    .then(generatedSwaggerString => {
+    .then(async generatedSwaggerString => {
       const oas = new OAS(generatedSwaggerString);
+      const validate = promisify(oas.validate);
```
Comment (erunion) [inline main.js:81]: Real promise support on this library is coming soon!

File: main.js [hunk 2]
```diff
@@ -105,7 +139,7 @@ async function run() {
       });
     })
     .catch(err => {
-      core.setFailed(`There was an error finding or loading your OAS file.\n\n${err.message || err}`);
+      core.setFailed(`There was an error finding or loading your OpenAPI/Swagger file.\n\n${err.message || err}`);
```
Comment (erunion) [inline main.js]: rdme has this similar error message and I've always wanted it to be contextual to what happened so if we borked an OAS it'd never mention Swagger.

**Review by erunion — COMMENTED** [review]
this all looks awesome.

do you also want to see if you can get the action working without having `node_modules/` stored in git? 🥺

**Comment by kanadgupta** [comment]
> do you also want to see if you can get the action working without having `node_modules/` stored in git? 🥺

@erunion yeah, I don't love that either... [the GitHub docs](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github:~:text=As%20an%20alternative%20to%20checking%20in,into%20one%20file%20used%20for%20distribution.) suggest using [`@vercel/ncc`](https://github.com/vercel/ncc) instead of `node_modules/`, which would have us commit a dist file instead. Should we do that?

**Comment by erunion** [comment]
@kanadgupta I'd be good with that or a Docker solution like in https://github.com/brpaz/action-semantic-release

https://github.com/brpaz/action-semantic-release/blob/master/Dockerfile
https://github.com/brpaz/action-semantic-release/blob/master/action.yml#L13-L15
https://github.com/brpaz/action-semantic-release/blob/master/entrypoint.sh

Any solution is probably out of scope for this PR, but whatever you want to do is cool with me.

**Review by mjcuva — APPROVED** [review]
I haven't tested, but this all seems good to me!

**Pull request merged**